### PR TITLE
Feature/rollups  reset crlps

### PR DESCRIPTION
**Pull request by bethbrains**
# Critical Changes

# Changes
- Changed CRLP Enablement to a toggle
- New layout on CRLP enablement page
- Added Reset CRLPs button that removes any new rollups, filter groups, and filter rules you created while using Customizable Rollups and resets all NPSP legacy rollups (including any User Defined Rollups) back to their default behavior

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/CRLP_DefaultConfigBuilder_SVC.cls [hunk 1]
```diff
@@ -626,10 +641,108 @@ public class CRLP_DefaultConfigBuilder_SVC {
...
+            Set<String> defaultGroupNames = new Set<String>();
+            for (CRLP_RollupCMT.FilterGroup grp : groups) {
+                defaultGroupNames.add(grp.recordName);
+                for (CRLP_RollupCMT.FilterRule rule : grp.rules) {
+                    defaultRuleNames.add(rule.recordName);
+                }
+            }
+
+            for (Filter_Rule__mdt rule : allExistingRules) {
+                if (!defaultRuleNames.contains(rule.DeveloperName) && defaultGroupNames.contains(rule.Filter_Group__r.DeveloperName)) {
```
Comment (randi274) [inline src/classes/CRLP_DefaultConfigBuilder_SVC.cls]: Did you consider just checking the start of the developer name for NPSP/NPSP_ instead?

File: src/pages/STG_PanelCustomizableRollup.page [hunk 2]
```diff
@@ -51,7 +49,7 @@
                                     </apex:outputText>
                                 </div>
                                 <div class="slds-media__body">
-                                    <h2 class="slds-text-heading_small slds-text-heading--small">
+                                    <h2 class="slds-text-heading_small slds-text-heading_small">
```
Comment (randi274) [inline src/pages/STG_PanelCustomizableRollup.page]: Can you also add the same underscore fix on line 26?

File: src/pages/STG_PanelCustomizableRollup.page [hunk 3]
```diff
@@ -61,52 +59,118 @@
...
+            <!-- BEGIN INTRO TEXT AND TOGGLE -->
+            <div class="slds-grid {!IF(isHHAccount, '', 'slds-hide')}">
+                <apex:outputPanel rendered="{! !isPolling }">
+                    <div class="slds-m-vertical_medium">
+                        <apex:outputText escape="false" styleClass="slds-text-body_small" value="{!$Label.stgHelpCustomizableRollupsEnable1}"/>
+                    </div>
+
+                    <label class="slds-checkbox_toggle slds-grid slds-m-vertical_small">
+                        <span class="slds-form-element__label slds-m-bottom_none">Customizable Rollups</span>
+                        <apex:inputCheckbox onClick="onToggle(); return false;" id="enableCRLPToggle" html-name="checkbox-toggle-2" value="{!stgService.stgCRLP.Customizable_Rollups_Enabled__c}" html-aria-describedby="checkbox-toggle-2" />
```
Comment (randi274) [inline src/pages/STG_PanelCustomizableRollup.page]: This checkbox isn't keyboard accessible when CRLPs is off, and when it's on it's actually navigating to the app. This looks like a problem with the apex:inputCheckbox, but maybe is a result of using onClick vs using onSelect?

Comment (randi274) [reply src/pages/STG_PanelCustomizableRollup.page]: This is working with space! Looks great.

File: src/classes/CRLP_DefaultConfigBuilder_SVC.cls (hunk 1 above)
Comment (bethbrains) [reply src/classes/CRLP_DefaultConfigBuilder_SVC.cls]: Yes but we didn't want to rely on a naming convention, so we went for the (slightly less performant) loop of the actual record names.

File: src/pages/STG_PanelCustomizableRollup.page (hunk 2 above)
Comment (randi274) [reply src/pages/STG_PanelCustomizableRollup.page]: Fixed

**Comment by randi274** [comment]
@dgamboni this dev-approved. We're going to rebase into Master to get around the feature/rollups extra work, so hold off on merging until the existing feature/rollups PR -> master is merged. We'll update when it is ready too. Let me or @bethbrains know if that needs any clarification!

**Comment by smarovich** [comment]
@bethbrains @dgamboni I updated the release notes for this PR. Let me know if you'd like to see any changes. TIA!

**Pull request merged**